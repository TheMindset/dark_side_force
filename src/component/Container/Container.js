import React from 'react'
import './Container.scss'
import PropTypes from 'prop-types'
import MovieCard from '../MovieCard/MovieCard'
import CharacterCard from '../CharacterCard/CharacterCard'

const Container = ({ cards }) => {
  const allCards = cards.map(card => {
    if (card.title) {
      return <MovieCard key={card.episode} movie={card} />
    }
      return <CharacterCard key={card.name} character={card}/>
  })

  return (
    <section className="container"> {allCards} </section>
  )
}

export default Container

Container.propTypes = {
  cards: PropTypes.array.isRequired
}
