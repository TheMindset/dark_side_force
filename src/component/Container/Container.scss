* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.container {
  background-attachment: fixed;
  background-image: url('../Images/container-background.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  padding: 80px;
}