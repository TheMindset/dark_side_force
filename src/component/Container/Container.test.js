import React from 'react'
import { shallow } from 'enzyme'
import Container from './Container'


describe('Container whithout favorites', () => {
  const mockFavorites = []
  const mockToggleFavorite = jest.fn()

  let wrapper

  test('should ', () => {
    wrapper = shallow(
      <Container
        cards={mockFavorites}
        type='favorites'
        favorites={mockFavorites}
        toggleFavorite={mockToggleFavorite}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Container with favorites', () => {
  const mockFavorites = [
    {
      name: 'Luke Skywalker',
      homeworld: 'Tattooine',
      population: 200000,
      species: 'Human',
      films: [ 'Terrible', 'Incroyable']
    },
    {
      name: 'Luke C3-PO',
      homeworld: 'Tattooine',
      population: 200000,
      species: 'Droid',
      films: ['All of them', 'Terrible']
    }
  ]
  const mockToggleFavorite = jest.fn()

  let wrapper

  test('should ', () => {
    wrapper = shallow(
      <Container
        cards={mockFavorites}
        type='favorites'
        favorites={mockFavorites}
        toggleFavorite={mockToggleFavorite}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Container with characters', () => {
  const mockCharacters = [
    {
      name: 'Captain America',
      homeworld: 'Terre',
      population: 200000000,
      species: 'Human',
      films: [ 'All Avengers', "It's crazy"]
    },
    {
      name: 'Thor',
      homeworld: 'Asgard',
      population: 20,
      species: 'Asgardian',
      films: ['All Avengers', 'So crazy man!!']
    }
  ]
  const mockToggleFavorite = jest.fn()

  let wrapper

  test('should ', () => {
    wrapper = shallow(
      <Container
        cards={mockCharacters}
        type='characters'
        toggleFavorite={mockToggleFavorite}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})


describe('Container with movies', () => {
  const mockMovies = [
    {
      title: 'The Avengers',
      episode: 1,
      releaseYear: 2012,
      id: 1
    },
    {
      title: 'Avengers: Age of Ultron',
      episode: 2,
      releaseYear: 2015,
      id: 2
    }
  ]
  const mockToggleFavorite = jest.fn()

  let wrapper

  test('should ', () => {
    wrapper = shallow(
      <Container
        cards={mockMovies}
        toggleFavorite={mockToggleFavorite}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
