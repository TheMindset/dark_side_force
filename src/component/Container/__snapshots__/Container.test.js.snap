// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container whithout favorites should  1`] = `
<div
  className="empty-favorites"
>
  <h3>
    Add some Favorites
  </h3>
</div>
`;

exports[`Container with characters should  1`] = `
<section
  className="container"
>
   
  <CharacterCard
    character={
      Object {
        "films": Array [
          "All Avengers",
          "It's crazy",
        ],
        "homeworld": "Terre",
        "name": "Captain America",
        "population": 200000000,
        "species": "Human",
      }
    }
    key="Captain America"
    toggleFavorite={[MockFunction]}
  />
  <CharacterCard
    character={
      Object {
        "films": Array [
          "All Avengers",
          "So crazy man!!",
        ],
        "homeworld": "Asgard",
        "name": "Thor",
        "population": 20,
        "species": "Asgardian",
      }
    }
    key="Thor"
    toggleFavorite={[MockFunction]}
  />
   
</section>
`;

exports[`Container with favorites should  1`] = `
<section
  className="container"
>
   
  <CharacterCard
    character={
      Object {
        "films": Array [
          "Terrible",
          "Incroyable",
        ],
        "homeworld": "Tattooine",
        "name": "Luke Skywalker",
        "population": 200000,
        "species": "Human",
      }
    }
    favorites={
      Array [
        Object {
          "films": Array [
            "Terrible",
            "Incroyable",
          ],
          "homeworld": "Tattooine",
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "Human",
        },
        Object {
          "films": Array [
            "All of them",
            "Terrible",
          ],
          "homeworld": "Tattooine",
          "name": "Luke C3-PO",
          "population": 200000,
          "species": "Droid",
        },
      ]
    }
    key="Luke Skywalker"
    toggleFavorite={[MockFunction]}
  />
  <CharacterCard
    character={
      Object {
        "films": Array [
          "All of them",
          "Terrible",
        ],
        "homeworld": "Tattooine",
        "name": "Luke C3-PO",
        "population": 200000,
        "species": "Droid",
      }
    }
    favorites={
      Array [
        Object {
          "films": Array [
            "Terrible",
            "Incroyable",
          ],
          "homeworld": "Tattooine",
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "Human",
        },
        Object {
          "films": Array [
            "All of them",
            "Terrible",
          ],
          "homeworld": "Tattooine",
          "name": "Luke C3-PO",
          "population": 200000,
          "species": "Droid",
        },
      ]
    }
    key="Luke C3-PO"
    toggleFavorite={[MockFunction]}
  />
   
</section>
`;

exports[`Container with movies should  1`] = `
<section
  className="container"
>
   
  <MovieCard
    key="1"
    movie={
      Object {
        "episode": 1,
        "id": 1,
        "releaseYear": 2012,
        "title": "The Avengers",
      }
    }
  />
  <MovieCard
    key="2"
    movie={
      Object {
        "episode": 2,
        "id": 2,
        "releaseYear": 2015,
        "title": "Avengers: Age of Ultron",
      }
    }
  />
   
</section>
`;
